server/src-lib/Hasura/GraphQL/Transport/HTTP.hs

```diff
@@ -72,10 +84,12 @@ runHasuraGQ
   -> GQLReqUnparsed
   -> UserInfo
   -> E.ExecOp
-  -> m EncJSON
+  -> m ((NominalDiffTime, Telem.QueryType), EncJSON)
```

Comment:
It seems strange to me to use a nested pair here instead of a 3-tuple. Is there a particular reason to do it this way?

Reply:
> Is there a particular reason to do it this way?

Laziness!